### DISCUSSION TITLE
How to run pyscript file with specific virtual environment?

### DISCUSSION THREAD

**Discussion by ELITSZORD** [teamalgoritma/community#819] [Q&A] [answered]
Dear Team Algo,

Tiap-tiap projek kan pasti membutuhkan virtual environment yang isi librarynya berbeda2 di setiap env. Bagaimana caranya ketika run pyscript dia jalan juga bersama virtual environment yang kita buat (misal anomaly)? 

Thanks.

Best Regards,

Faisal

**Comment by arkanayudhistira** [comment]
Dear @ELITSZORD,

Jika anda menggunakan Visual Studio Code untuk menjalanklan file .py, silahkan lakukan step berikut:

1. Buka folder .py script yang akan anda jalankan melalui **File** -> **Open Folder** pada VS Code 
2. Buka **Terminal** -> **New Terminal** pada VS Code
3. Pilih **Command Prompt** sebagai jenis terminal yang akan digunakan
4. Aktivasi Virtual Environment yang akan anda gunakan, dengan menjalankan
    ```
    conda activate <nama environment>
    ```
5. Jalankan aplikasi anda dengan menjalankan
    ```
    python run <nama app .py>
    ```

Semoga membantu!

**Comment by ELITSZORD** [comment]
Dear @arkanayudhistira ,

Maksudnya bukan begitu mas, kita kan run pyscript menggunakan batch file dan task scheduller. Tp ketika kita run pyscript tanpa vscode itu tidak menggunakan virtual environment yang kita buat. Inginnya otomatis ketika file pyscript di run tanpa vscode itu virtual environment yang kita inginkan juga jalan brng. Itu bagaimana ya mas?

**Reply by hwulanayu** [reply, marked as answer]
Dear @ELITSZORD,

Ada 2 cara yang dapat dilakukan untuk menjalankan sebuah pyscript dengan virtual environment yang spesifik:

1. Membuat script python yang mengaktifkan virtual env terkait dan running script yang diinginkan.
2. Menambahkan path virtual environment tersebut di script yang sudah ada

Bapak bisa merefer ke link [stackoverflow](https://stackoverflow.com/questions/6943208/activate-a-virtualenv-with-a-python-script) ini untuk kodenya. 

Demikian jawaban yang dapat kami sampaikan. Semoga membantu! Jika ternyata jawaban ini menyelesaikan permasalahan Anda, silakan _mark as anwer._

**Reply by IrfanChairurrachman** [reply]
Halo @ELITSZORD , karena tidak ada konfirmasi lebih lanjut lebih dari 7 hari dan untuk menjaga agar forum diskusi tetap berjalan kondusif, kami akan melakukan lock/close diskusi ini. Semoga dapat dipahami.
